src/MusicStore/Controllers/CheckoutController.cs

```diff
@@ -78,17 +88,24 @@ public IActionResult AddressAndPayment()
             [FromServices] MusicStoreContext dbContext,
             int id)
         {
+            var userName = HttpContext.User.Identity.Name;
+
             // Validate customer owns this order
             bool isValid = await dbContext.Orders.AnyAsync(
                 o => o.OrderId == id &&
-                o.Username == HttpContext.User.Identity.Name);
+                o.Username == userName);
 
             if (isValid)
             {
+                _logger.LogInformation("User {userName} completed checkout on order {orderId}.", userName, id);
```

Comment:
You mean [these](https://github.com/aspnet/Mvc/blob/a78f77afde003c4a3fcf5dd7b6dc13dd9c85f825/src/Microsoft.AspNetCore.Mvc.Formatters.Json/Internal/MvcJsonLoggerExtensions.cs#L17)? He specifically said not to use those to keep it simple.